src/components/header/components/NavItem.tsx

```diff
@@ -3,22 +3,32 @@ import classNames from 'classnames';
 import { ChevronRight as ChevronRightIcon } from '../../icons';
 
 export interface NavItemProps extends HTMLProps<HTMLAnchorElement> {
+  asElement?: React.ElementType;
+  to?: string;
```

Comment:
Could we potentially move the `asElement` and `to` declarations into a shared interface somewhere else in the codebase (i.e. maybe a new file in `util/types`)? I was thinking something like:
```ts
interface AsElementLink<T extends HTMLElement> extends HTMLProps<T> {
    asElement?: React.ElementType;
    to?: string;
}
```

Reply:
Refactored to pull out the interface into a separate file